app/models/embed/purl/resource_file.rb

```diff
@@ -40,15 +52,34 @@ def stacks_url
       end
 
       def label_or_filename
+        return caption_label if caption?
+
         label.presence || filename
       end
```

Comment:
@jcoyne I know we discussed putting the caption label logic in the view layer, but the RF instance already seems to has some logic around what label to present to the world (in `#label_or_filename`). I could move the caption label logic *back* into the component, but then we're overriding what the RF asserts as its label or filename which strikes me as something we don't do elsewhere. Happy to be convinced, though, so we can get this work moving.

Reply:
This seems to fine to me.